RFM9x.device.nut

```diff
@@ -0,0 +1,404 @@
+        rf.setMode(RFM9X_RXCONTINUOUS);
+    }
+    
+    function setPayloadLength(len) {
+        _writeReg(RFM9X_REG_PAYLOAD_LENGTH, len);
+    }
+    
+    function setImplicitHeaderMode(state) {
+        local cur = _readReg(RFM9X_REG_MODEM_CONFIG1);
+        _writeReg(RFM9X_REG_MODEM_CONFIG1, (cur & 0xfe) | (state ? 1 : 0));
+    }
+    
+    function setPreambleLength(len) {
+        _writeReg(RFM9X_REG_PREAMBLE_MSB, (len >>8) & 0xff);
+        _writeReg(RFM9X_REG_PREAMBLE_LSB, len & 0xff);
+    }
+    
+    function setFifoTxBase(start) {
+        _writeReg(RFM9X_REG_FIFO_TX_BASE_ADDR, start & 0xff);
+    }
+    
+    function setFifoRxBase(start) {
+        _writeReg(RFM9X_REG_FIFO_RX_BASE_ADDR, start & 0xff);
+    }
+    
+    function setCodingRate(cr) {
+        local cur = _readReg(RFM9X_REG_MODEM_CONFIG1);
+        local clear = cur & 0xf1; // exclude bits 3-1
+        _writeReg(RFM9X_REG_MODEM_CONFIG1, clear | (CRTABLE[cr] << 1));
+    }
+    
+    function setRxPayloadCRC(state) {
+        local current = _readReg(RFM9X_REG_MODEM_CONFIG2);
+        _writeReg(RFM9X_REG_MODEM_CONFIG2, (current & 0xfb) | (state ? 0x04 : 0x00));
+    }
+    
+    function setBandwidth(bw) {
+        local current = _readReg(RFM9X_REG_MODEM_CONFIG1);
+        _writeReg(RFM9X_REG_MODEM_CONFIG1, BWTABLE[bw] << 4 | (current & 0x0f));
+    }
+    
+    function setMode(mode) {
+        local current = _readReg(RFM9X_REG_OP_MODE);
+        _writeReg(RFM9X_REG_OP_MODE, mode | (current & 0xf8));
+    }
+        
+    function clearInterrupts() {
+        _writeReg(RFM9X_REG_IRQ_FLAGS, 0xff);
+    }
+    
+    function maskAllInterrupts() {
+        _writeReg(RFM9X_REG_IRQ_FLAGS_MASK, 0xff);
+    }
+    
+    // Enable a specific interrupt. Avaliable interrupts are in
+    // RFM9X_FLAGS
+    function enableInterrupt(mask) {
+        local currentInterrupts = _readReg(RFM9X_REG_IRQ_FLAGS);
+        // Mask the interrupt by writing high, therefore enable by writing low
+        local newInterrupts = currentInterrupts & (~ (1<<mask));
+        _writeReg(RFM9X_REG_IRQ_FLAGS_MASK, newInterrupts);
+    }
+
+    function setMaxPayload(pl) {
+        pl = pl & 0xff;
+        _writeReg(RFM9X_REG_MAX_PAYLOAD_LENGTH, pl);
+    }
+    
+    function setSpreadingFactor(sf) {
+        _writeReg(RFM9X_REG_MODEM_CONFIG2, (sf & 0x0f) << 4);
+    }
+    
+    function setFrequency(freq) {
+        setMode(RFM9X_SLEEP);
+        local reg_freq = (freq/RFM9X_FREQ_STEP).tointeger();
+        local freq_blob = blob(4); // big-endian blob
+        freq_blob[0] = RFM9X_REG_FREQUENCY_HIGH | RFM9X_WRITE_MASK;
+        freq_blob[1] = (reg_freq >> 16) & 0xFF;
+        freq_blob[2] = (reg_freq >> 8) & 0xFF;
+        freq_blob[3] = (reg_freq) & 0xFF;
+        
+        _csLow();
+        _spiModule.writeread(freq_blob);
+        _csHigh();
+    }
+
+    function setReceiveHandler(handler) {
+        _receiveHandler = handler;
+    }
+
+    function isDoneSending() {
```

Comment:
I would just revert the conditions:

1. `_sendFinished ` -> `_isSending`
2. `isDoneSending ` -> `isSending`